### PR TITLE
Reverting #12659 from 24.0 release

### PR DESCRIPTION
**Pull request by abhishekagarwal87**
https://github.com/apache/druid/pull/13027 highlights an issue with current filter matching that was introduced in #12659. Since tackling that can take time, I am going to revert this change from 24 release. So we don't have to worry about backward compatibility as we find the right solution.

### CODE REVIEW

**Review by kfaraz — APPROVED** [review]
LGTM

File: docs/ingestion/native-batch-input-source.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ For general information on native batch indexing and parallel task indexing, see
 
 ## S3 input source
 
-> You need to include the [`druid-s3-extensions`](../development/extensions-core/s3.md) as an extension to use the S3 input source.
+> You need to include the [`druid-s3-extensions`](../development/extensions-core/s3.md) as an extension to use the S3 input source.
```
Comment (vogievetsky) [inline docs/ingestion/native-batch-input-source.md:32]: This change is a typo right? There is no need for the trailing space.

**Comment by vogievetsky** [comment]
I think you also want to remove it from the web console. You need to remove the lines:

https://github.com/apache/druid/blob/master/web-console/src/druid-models/ingestion-spec/ingestion-spec.tsx#L654
https://github.com/apache/druid/blob/master/web-console/src/druid-models/ingestion-spec/ingestion-spec.tsx#L816
https://github.com/apache/druid/blob/master/web-console/src/druid-models/ingestion-spec/ingestion-spec.tsx#L875
https://github.com/apache/druid/blob/master/web-console/src/druid-models/input-source/input-source.tsx#L437

File: docs/ingestion/native-batch-input-source.md (hunk 1 above)
Comment (kfaraz) [reply docs/ingestion/native-batch-input-source.md:32]: Yeah, the trailing space was removed in the original PR. Reverting that commit would have brought back this change.

**Comment by abhishekagarwal87** [comment]
thanks for catching that @vogievetsky. I have removed them in latest commit.

**Review by vogievetsky — APPROVED** [review]

**Comment by vogievetsky** [comment]
the alert about `const inputSourceFilter: Field<IoConfig> = {` not being needed is obviously a nothing burger but it might fail CI due to our tslint validation

**Comment by vogievetsky** [comment]
Ok I can confirm that it will fail. But if you remove https://github.com/abhishekagarwal87/druid/blob/revert_glob_filter_2/web-console/src/druid-models/ingestion-spec/ingestion-spec.tsx#L457 then it will pass all web console tests (unit and e2e), just confirmed locally

**Comment by vogievetsky** [comment]
closing this PR in favor of: https://github.com/apache/druid/pull/13046

**Pull request closed without merging**